### PR TITLE
Docs: Translation memory reworked

### LINKED ISSUE #8288: Update Translation Memory documentation about rebuild, etc.
### Describe the problem

Translation memory management has evolved in recent versions, and the documentation is not reflecting that.

### Describe the solution you'd like

Describe all features of translation memory management within UI, including on the project level. Like the rebuild button!
Introduced via **https://github.com/WeblateOrg/weblate/issues/7347**

### PR DESCRIPTION
**Pull request by comradekingu**
Closes #8288
## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [ ] I have described the changes in the commit messages.

## Other information

No screenshots yet.

### CODE REVIEW

File: docs/admin/memory.rst [hunk 1]
```diff
@@ -74,30 +73,64 @@ Managing translation memory
...
 :djadmin:`dump_memory`
     Exports the memory into JSON
 :djadmin:`import_memory`
     Imports TMX or JSON files into the translation memory
+
+.. versionadded:: 4.14
+
+The Weblate API covers the translation memory.
+This allows automating manipulation for different purposes,
+or based on events in the translation cycle.
```
Comment (comradekingu) [inline docs/admin/memory.rst:136]: This part is guesswork that could possibly be improved by some realistic scenarios.

File: docs/admin/memory.rst [hunk 2]
```diff
@@ -24,29 +24,28 @@ turned on by default.
...
 -------------------------
 
 .. versionadded:: 3.2
 
    In earlier versions translation memory could be only loaded from a file
    corresponding to the current imported translation memory scope.
 
-The translation memory scopes are there to allow both privacy and sharing of
-translations, to suit the desired behavior.
+The translation memory scopes allow both privacy and sharing translations.
```
Comment (comradekingu) [inline docs/admin/memory.rst:35]:
```suggestion
The translation memory scopes sharing translations with only relevant components, and projects, or for the entire platform.
```
Does it allow sharing between teams, users, or something else too?

File: docs/admin/memory.rst [hunk 3]
```diff
@@ -24,29 +24,28 @@ turned on by default.
 
 .. _memory-scopes:
 
-Translation memory scopes
+Translation-memory scopes
```
Comment (nijel) [inline docs/admin/memory.rst:27]: Nobody calls it this way.

https://en.wikipedia.org/wiki/Translation_memory, https://omegat.org/, https://www.trados.com/solutions/translation-memory/

File: docs/admin/memory.rst (hunk 2 above)
Comment (orangesunny) [reply docs/admin/memory.rst:35]: This part of the article describes only that there are several scopes. Their feature are described further down, so it is not needed to list it here.

Comment (comradekingu) [reply docs/admin/memory.rst:35]: It is quite difficult to understand what "translation memory scopes" are.
Something that allows for privacy, and sharing translations somehow doesn't clear that up.

File: docs/admin/memory.rst [hunk 4]
```diff
@@ -32,7 +32,7 @@ Translation memory scopes
    In earlier versions translation memory could be only loaded from a file
    corresponding to the current imported translation memory scope.
 
-The translation memory scopes allow both privacy and sharing translations.
+The translation memory scopes ensure privacy, while enabling sharing translations.
```
Comment (comradekingu) [inline docs/admin/memory.rst:35]:
```suggestion
The translation memory scopes allow for privacy when sharing translations.
```
Privacy is never ensured, as that is a dangerous claim.
Having any privacy isn't a product of turning on the translation memory sharing.
This makes sense to me:
`Sharing the translation memory has fine-tuned settings to adjust the scope of how and where sharing occurs.`

**Review by orangesunny — APPROVED** [review]
good for this stage

File: docs/admin/memory.rst [hunk 5]
```diff
@@ -24,7 +24,7 @@ turned on by default.
 
 .. _memory-scopes:
 
-Translation-memory scopes
+Translation memory scopes
```
[the remark this answers is not in the document]
Comment (orangesunny) [reply docs/admin/memory.rst]: I don’t have the time or will to spend more of any on this.

File: docs/admin/memory.rst (hunk 4 above)
Comment (orangesunny) [reply docs/admin/memory.rst]: You are connecting here two things into one. There is a choice to keep TM private or share it. Together it won’t go well.

Comment (orangesunny) [reply docs/admin/memory.rst]: made the change to:

```
The translation-memory scopes ensure privacy for different projects and users. 
Sharing of translations is also available.
```

**Comment by orangesunny** [comment]
Let’s merge this in the current state, and follow up with polishing in https://github.com/WeblateOrg/weblate/issues/9555 if needed.

**Pull request merged**